library/lua/time.lua

```diff
@@ -0,0 +1,87 @@
+local _ENV = mkmodule('time')
+
+local TU_PER_DAY = 1200
+--[[
+if advmode then TU_PER_DAY = 86400 ? or only for cur_year_tick?
+advmod_TU / 72 = ticks
+--]]
+local TU_PER_MONTH = TU_PER_DAY * 28
+local TU_PER_YEAR = TU_PER_MONTH * 12
+
+local MONTHS = {
+ 'Granite',
+ 'Slate',
+ 'Felsite',
+ 'Hematite',
+ 'Malachite',
+ 'Galena',
+ 'Limestone',
+ 'Sandstone',
+ 'Timber',
+ 'Moonstone',
+ 'Opal',
+ 'Obsidian',
+}
+
+Time = defclass(Time)
+function Time:init(args)
+ self.year = args.year or 0
+ self.ticks = args.ticks or 0
+end
```

Comment:
```suggestion
Time.ATTRS{
    year=0,
    ticks=0,
}
```

Reply:
though it would still be useful to keep the `init` function so you could normalize the values if ticks is < 0 or > TU_PER_YEAR